src/rpc/blockchain.cpp

```diff
@@ -1337,7 +1337,7 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
     }
 
     // orchard
-    {
+    if (Params().GetConsensus().NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_NU5)) {
```

Comment:
Comment on line 1351 and the subsequent loop: why would a post-NU5 block not have an Orchard final state?

Reply:
That's a good question; the same applies to line 1325. I'm not sure why blocks might not have a final state after activation.

Reply:
Oh! It might be defensive against the eventual deactivation of these protocols; that's the only situation under which I'd expect there to be a series of blocks *after* a non-null root. But in that circumstance, I don't think we'd want to scan. I'd like to get @str4d's comments here to decide what to do about this.

The tests for this code explicitly check for a null `finalRoot` value, so rather than omitting the section entirely I'll just skip the expensive scan.